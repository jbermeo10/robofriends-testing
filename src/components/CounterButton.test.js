import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from './CounterButton';

it('se espera renderizar el componente CounterButton', () => {
  const mockColor = 'red'
  expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
})

it('incrementa correctamente el contador', () => {
  const mockColor = 'red'
  const wrapper = shallow(<CounterButton color={mockColor} />);

  wrapper.find('[id="counter"]').simulate('click');
  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 2 });
  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 3 });
  wrapper.find('[id="counter"]').simulate('keypress');
  expect(wrapper.state()).toEqual({ count: 3 });

  // expect(wrapper.props().color).toEqual({ color: 'red' });
  expect(wrapper.props().color).toEqual('red');
})