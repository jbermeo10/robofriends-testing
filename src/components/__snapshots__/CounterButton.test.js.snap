// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`se espera renderizar el componente CounterButton 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Cuenta de clicks: 
  0
</button>
`;
